Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_browser_driver.py

```diff
@@ -41,6 +42,42 @@ def restore_env_after_all_testing(self):
     def close_browsers(self):
         self._terminate_processes(self.process_name, self.bundle_id)
 
+    @contextmanager
+    def profile(self, output_path, profile_filename, profiling_interval, trace_type='profile', timeout=300):
+        trace_process = None
+        additional_trace_arguments = {
+            'full': ['--type=full'],
+            'profile': ['--type=profile'],
+        }
+        if trace_type not in additional_trace_arguments:
+            _log.error('`{}` is not a valid type of trace. Defaulting to `profile`.'.format(trace_type))
+            trace_type = 'profile'
+        try:
+            _log.info('Gathering traces')
+            trace_command = ['sudo', 'ktrace', 'artrace', '-o', os.path.join(output_path, profile_filename + '.ktrace')]
+            trace_command += additional_trace_arguments[trace_type]
+            if profiling_interval:
+                trace_command += ['-i', profiling_interval]
+            _log.info('Running ktrace command: {}'.format(trace_command))
+            trace_process = subprocess.Popen(trace_command)
+            time.sleep(5)  # Wait a few seconds for `ktrace` process to start
+            yield
+        except Exception as error:
+            raise Exception('Failed to start ktrace. Error: {}'.format(error))
+        finally:
+            if not trace_process:
+                return
+            try:
+                subprocess.call(['sudo', 'pkill', '-2', 'ktrace'])
+                _log.info('Stopping ktrace task.')
+                with Timeout(timeout):
+                    trace_process.wait()
+            except Exception as error:
+                _log.error('Failed to quit ktrace.')
+            finally:
+                _log.info('Killing ktrace task.')
+                subprocess.call(['sudo', 'pkill', '-9', 'ktrace'])
```

Comment:
Ditto

Reply:
I've tried the following:
`trace_process.send_signal(signal.SIGINT)`
`os.kill(trace_process.pid, signal.SIGINT)`
`subprocess.call(['sudo', 'kill', '-2', str(trace_process.pid)])`
None of these seem to work - only `pkill` does. My suspicion is that there are multiple processes called `ktrace` being spawned, and all of them need to receive SIGINT in order for the trace to end.